apollo-api/src/main/java/com/apollographql/apollo/api/internal/json/InputFieldJsonWriter.kt

```diff
@@ -0,0 +1,222 @@
+package com.apollographql.apollo.api.internal.json
+
+import com.apollographql.apollo.api.CustomTypeValue.*
+import com.apollographql.apollo.api.ScalarType
+import com.apollographql.apollo.api.ScalarTypeAdapters
+import com.apollographql.apollo.api.internal.InputFieldMarshaller
+import com.apollographql.apollo.api.internal.InputFieldWriter
+import com.apollographql.apollo.api.internal.json.Utils.writeToJson
+import java.io.IOException
+
+class InputFieldJsonWriter(
+    private val jsonWriter: JsonWriter,
+    private val scalarTypeAdapters: ScalarTypeAdapters
+) : InputFieldWriter {
+
+  @Throws(IOException::class)
+  override fun writeString(fieldName: String, value: String?) {
+    if (value != null) {
```

Comment:
For consistency, maybe we can invert this conditional and have `value == null` first.